src/lib/connector.ts

```diff
@@ -212,21 +158,15 @@ export class Web3AuthConnector extends Connector {
 
   async getChainId(): Promise<number> {
     try {
-      const provider = await this.getProvider();
-      if (!provider) {
-        const networkOptions = this.socialLoginAdapter.chainConfigProxy;
-        if (typeof networkOptions === "object") {
-          const chainID = networkOptions.chainId;
-          if (chainID) {
-            return normalizeChainId(chainID);
-          }
-        }
-      } else {
-        const chainId = await provider.request({ method: "eth_chainId" });
+      if (this.provider) {
+        const chainId = await this.provider.request({ method: "eth_chainId" });
         if (chainId) {
           return normalizeChainId(chainId as string);
         }
       }
+      if (this.initialChainId) {
```

Comment:
why this?

Reply:
The `getChainId` function is triggered even before Web3Auth connects, so when the provider doesn't exist it throws an error and gets stuck. We I added this as a fix that it can return the chain id for the first chain directly.